### ISSUE TITLE
absolute_url twig function does not work properly from CLI

### ISSUE THREAD

**Issue by alsma** [symfony/symfony#17050]
Hi! 

`absolute_url` does rely on request stack, and because of this it's impossible to generate absolute url for asset in current version from CLI.

Before it worked fine via [this extension](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/TwigBundle/Extension/AssetsExtension.php#L108). 

I see only one way to do this through url packages, but it looks bad from my POV.
My use case is sending email with HTML body from console command.

Thanks in advance!

**Comment by alsma** [comment]
> I see only one way to do this through url packages
Here is a workaround that i'm talking about 
```
framework:
    assets:
        packages:
            mail:
                base_urls: ['%router.request_context.scheme%://%router.request_context.host%']
```

then use as 
```
{{ asset('content/images/mail/twitter@2x.png', 'mail') }}
```

**Comment by xabbuh** [comment]
That's unfortunately a known issue for which we need to find a fix (see #15448).

**Comment by patrickbussmann** [comment]
this is still an issue in 4.3.

    {{ absolute_url(asset('assets/images/header.png')) }}

results in

    http://localhost/assets/images/header.png

But even the settings as the request url contains a path.

    POST http://localhost/example/web/public/api/resetting/request

I expect something like

    http://localhost/example/web/public/assets/images/header.png
